DSharpPlus.SlashCommands/SlashCommandsExtension.cs

```diff
@@ -479,33 +509,28 @@ private static List<DiscordApplicationCommandOptionChoice> GetChoiceAttributesFr
         }
 
         //Small method to get the parameter's type from its type
-        private ApplicationCommandOptionType GetParameterType(Type type)
+        private ApplicationCommandOptionType GetParameterType<T>()
         {
-            if (type == typeof(string))
-                return ApplicationCommandOptionType.String;
-            if (type == typeof(long) || type == typeof(long?))
-                return ApplicationCommandOptionType.Integer;
-            if (type == typeof(bool) || type == typeof(bool?))
-                return ApplicationCommandOptionType.Boolean;
-            if (type == typeof(double) || type == typeof(double?))
-                return ApplicationCommandOptionType.Number;
-            if (type == typeof(DiscordChannel))
-                return ApplicationCommandOptionType.Channel;
-            if (type == typeof(DiscordUser))
-                return ApplicationCommandOptionType.User;
-            if (type == typeof(DiscordRole))
-                return ApplicationCommandOptionType.Role;
-            if (type == typeof(DiscordEmoji))
-                return ApplicationCommandOptionType.String;
-            if (type == typeof(TimeSpan?))
-                return ApplicationCommandOptionType.String;
-            if (type == typeof(SnowflakeObject))
-                return ApplicationCommandOptionType.Mentionable;
+            var type = typeof(T);
             if (type.IsEnum || Nullable.GetUnderlyingType(type)?.IsEnum == true)
                 return ApplicationCommandOptionType.String;
-            if (type == typeof(DiscordAttachment))
-                return ApplicationCommandOptionType.Attachment;
-            throw new ArgumentException("Cannot convert type! Argument types must be string, long, bool, double, TimeSpan?, DiscordChannel, DiscordUser, DiscordRole, DiscordEmoji, DiscordAttachment, SnowflakeObject, or an Enum.");
+            if (!ArgumentConverters.Any(converter => converter.Key == type))
+                throw new ArgumentException(
+                    @$"Could not convert type of argument. Provided {type.Name}, valid argument types are: {string.Join(", ", ArgumentConverters.Select(c => c.Key.IsGenericType &&
```

Comment:
This line is particularly long; consider using braces and initializing some variables